lib/js-yaml/loader.js

```diff
@@ -289,6 +289,11 @@ function storeMappingPair(state, _result, overridableKeys, keyTag, keyNode, valu
   // We need to convert keyNode to a string, but doing so can hang the process
   // (deeply nested arrays that explode exponentially using aliases) or execute
   // code via toString.
+
+  if (typeof keyNode === 'object') {
```

Comment:
Looks fine to me but https://stackoverflow.com/questions/8511281/check-if-a-value-is-an-object-in-javascript is interesting SO on this topic

Reply:
yeah i think i'll keep it this way, according to this post object covers objects and null and i think those would be the only two cases that would break things?

Reply:
Yes. I think it's fine, just wanted to point out that it's not a simple problem.

Reply:
ah for sure